src/js/Main/HistoryVideoWrapper.js

```diff
@@ -0,0 +1,155 @@
+import {
+  LOCAL_STORAGE_KEY,
+  CLASSNAME,
+  SNACKBAR_TEXT,
+} from "../constants/index.js";
+import { messenger, MESSAGE } from "../messenger/index.js";
+import { $, showSnackbar } from "../utils/index.js";
+import { HISTORY_VIDEO_TEMPLATE, renderMainVideo } from "../Video/index.js";
+
+export default class HistoryVideoWrapper {
+  constructor() {
+    this.initializeVariables();
+    this.selectHTMLElements();
+    this.addMessageListeners();
+    this.addEventListeners();
+    this.render();
+  }
+
+  initializeVariables() {
+    this.historyVideoItemsMap = new Map(
+      JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY.HISTORY_VIDEO_ITEMS))
+    );
+
+    this.historyVideosMap = new Map();
+  }
+
+  selectHTMLElements() {
+    this.$noSavedVideoImage = $(`.${CLASSNAME.NO_WATCHED_VIDEO_IMAGE}`);
+    this.$historyVideoWrapper = $(`.${CLASSNAME.HISTORY_VIDEO_WRAPPER}`);
+  }
+
+  addMessageListeners() {
+    messenger.addMessageListener(
+      MESSAGE.CANCEL_VIDEO_BUTTON_CLICKED,
+      this.deleteVideo.bind(this)
+    );
+
+    messenger.addMessageListener(
+      MESSAGE.MOVE_TO_HISTORY_ICON_CLICKED,
+      this.saveVideo.bind(this)
+    );
+
+    messenger.addMessageListener(
+      MESSAGE.HIDE_IF_VIDEO_IS_SAVED,
+      this.hideIfVideoIsSaved.bind(this)
+    );
+  }
+
+  addEventListeners() {
+    this.$historyVideoWrapper.addEventListener(
+      "click",
+      this.handleVideoWrapperClick.bind(this)
+    );
+  }
+
+  handleVideoWrapperClick(event) {
+    const { target: $videoWrapper } = event;
```

Comment:
```suggestion
    const $videoWrapper = event.target;
```

이럴때는 destructuring이 오히려 가독성을 해치는 것 같아요

Reply:
해당 부분은 리팩토링때 제거되었습니다.